pkg/lifecycle/render/render.go

```diff
@@ -58,16 +59,23 @@ func (r *renderer) Execute(ctx context.Context, release *api.Release, step *api.
 
 	r.StatusReceiver.SetProgress(ProgressRender)
 
+	// this should probably happen even higher up, like to the validation stage where we assign IDs to lifecycle steps, but moving it up here for now
+	if step.Root == "" {
+		step.Root = constants.InstallerPrefixPath
+	}
+
 	debug.Log("event", "render.plan")
 	pln, err := r.Planner.Build(step.Root, release.Spec.Assets.V1, release.Spec.Config.V1, release.Metadata, templateContext)
 	if err != nil {
 		return errors.Wrap(err, "build plan")
 	}
 
-	debug.Log("event", "backup.start")
-	err = r.backupIfPresent(constants.InstallerPrefixPath)
-	if err != nil {
-		return errors.Wrapf(err, "backup existing install directory %s", constants.InstallerPrefixPath)
+	if step.Root != "." && step.Root != "./" {
+		debug.Log("event", "backup.start")
+		err = r.backupIfPresent(step.Root)
+		if err != nil {
+			return errors.Wrapf(err, "backup existing install directory %s", constants.InstallerPrefixPath)
+		}
```

Comment:
Just a thought - should we check for, and disallow, the use of `..` within the render root?

Reply:
argh yeah. We probably need a proper chroot here someday.